File: tests/test_orthography_lacinka.py
"""Shared Python/JavaScript examples across both conversion layers."""

import json
from pathlib import Path

import pytest

from pravapis.pipeline import Converter
from pravapis.translit import Transliterator
from pravapis.types import Orthography, Script

CASES = json.loads(
    (Path(__file__).parent / "fixtures" / "orthography_lacinka.json").read_text(encoding="utf-8")
)


@pytest.mark.parametrize(("nark", "tarask", "latin"), CASES)
def test_conversion_and_lacinka_roundtrip(
    converter: Converter, data_dir: Path, nark: str, tarask: str, latin: str
) -> None:
    forward = Transliterator.load(Script.LACINKA, data_dir=data_dir)
    reverse = Transliterator.load(Script.LACINKA, reverse=True, data_dir=data_dir)
    for case in (str.lower, str.title, str.upper):
        classical = converter.convert(case(nark), Orthography.TARASKIEVICA).text
        assert classical == case(tarask)
        encoded = forward.transliterate(classical).text
        assert encoded == case(latin)
        decoded = reverse.transliterate(encoded).text
        assert decoded == classical
        assert converter.convert(decoded, Orthography.NARKAMAUKA).text == case(nark)

File: js/scripts/test-orthography-lacinka.ts
import assert from "node:assert/strict";
import { test } from "node:test";
import cases from "../../tests/fixtures/orthography_lacinka.json" with { type: "json" };
import { convert } from "../src/index.js";
import { transliterate } from "../src/translit.js";

for (const [nark, tarask, latin] of cases as [string, string, string][]) {
  test(`${nark} → ${tarask} → ${latin} and back`, () => {
    for (const recase of [
      (s: string) => s,
      (s: string) => s[0]!.toUpperCase() + s.slice(1),
      (s: string) => s.toUpperCase(),
    ]) {
      const classical = String(convert(recase(nark), { to: "taraskievica" }).text);
      assert.equal(classical, recase(tarask));
      const encoded = transliterate(classical, "lacinka");
      assert.equal(encoded, recase(latin));
      assert.doesNotMatch(encoded, /[\u0400-\u04ff]/u);
      const decoded = transliterate(encoded, "lacinka", { reverse: true });
      assert.equal(decoded, classical);
      assert.equal(convert(decoded, { to: "narkamauka" }).text, recase(nark));
    }
  });
}
